device/edison.go

```diff
@@ -99,48 +100,70 @@ func setConfig() error {
 	// get IP
 	var ip string
 
-	i := dialogs.SelectOneDialog("Chose the edison's inteface to connect via usb: ", []string{"Default", "Enter IP"})
+	i := dialogs.SelectOneDialog("Choose Edison's interface to connect to: ", []string{"Select from a list of interfaces", "Enter IP manually"})
 	fallback := false
 
 	if i == 0 {
 		ifaces, err := help.LocalIfaces()
 
 		if err != nil || len(ifaces) == 0 {
-			logrus.Error(err)
+			log.Error(err)
 			fmt.Println("[-] ", err.Error())
 			fallback = true
 		}
 
 		if !fallback {
 			arr := make([]string, len(ifaces))
 			arrSel := make([]string, len(ifaces))
+			fmt.Println("[+] Highlighted interfaces is our heuristic guess")
 			for i, iface := range ifaces {
 				arr[i] = iface.Name
 				arrSel[i] = iface.Name
 				if iface.Ipv4[:4] == "169." {
-					arrSel[i] = "\x1b[34m" + iface.Name + "\x1b[0m"
-					fmt.Println("[+] The highlighted interface is our heuristic guess")
+					if runtime.GOOS == "windows" {
+						arrSel[i] = "**" + iface.Name + "**"
+					} else {
+						arrSel[i] = "\x1b[34m" + iface.Name + "\x1b[0m"
+					}
 				}
 			}
 
 			i = dialogs.SelectOneDialog("Please chose correct interface: ", arrSel)
-			fmt.Println("[+] NOTE: You might need to provide your sudo password")
-			if out, err := help.ExecSudo(sudo.InputMaskedPassword, nil, "ifconfig", arr[i], "192.168.2.2"); err != nil {
-				fmt.Println("[-] Error running \x1b[34msudo ifconfig ", arrSel[i], " 192.168.2.2\x1b[0m: ", out)
-				fallback = true
+			if runtime.GOOS == "windows" {
+				// it's either: netsh interface ipv4 add address “Local Area Connection” 192.168.1.2 255.255.255.0
+				// or: netsh int ipv4 set address "%interface%" static %IP% %MASK% %GATE% gwmetric=1
+				command := fmt.Sprintf(`netsh int ipv4 set address "%s" static 192.168.2.2 255.255.255.0 192.168.2.1 gwmetric=1`, arr[i])
+				fmt.Println("[+] NOTE: You need to run this tool as an administrator")
+				if out, err := exec.Command(command).CombinedOutput(); err != nil {
+					fmt.Println("[-] Error running '", command, "': ", out)
+					fallback = true
+				}
+			} else {
+				fmt.Println("[+] NOTE: You might need to provide your sudo password")
+				if out, err := help.ExecSudo(sudo.InputMaskedPassword, nil, "ifconfig", arr[i], "192.168.2.2"); err != nil {
+					fmt.Println("[-] Error running 'sudo ifconfig ", arrSel[i], " 192.168.2.2': ", out)
+					fallback = true
+				}
+
 			}
 
 			ip = "192.168.2.15"
 		}
 	}
 
 	if i == 1 || fallback {
-		fmt.Println("NOTE: You might need to run `sudo ifconfig {interface} \x1b[34m192.168.2.2\x1b[0m` in order to access Edison at \x1b[34m192.168.2.15\x1b[0m")
+		if runtime.GOOS == "windows" {
```

Comment:
well, it's not about higher level. It's about putting string "windows" inside constant named Windows or something. It's just doesn't make sense to me.